sdk/typespec/typespec_client_core/src/http/policies/logging.rs

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+use crate::http::{
+    policies::{Policy, PolicyResult},
+    sanitizer::{Sanitizer, DEFAULT_ALLOWED_HEADER_NAMES, DEFAULT_ALLOWED_QUERY_PARAMETERS},
+    Context, LoggingOptions, Request,
+};
+use std::collections::HashSet;
+use std::sync::Arc;
+use tracing::info;
+
+/// [`Policy`] to log  a request and response.
+#[derive(Clone, Debug, Default)]
+pub struct LoggingPolicy {
+    allowed_headers: HashSet<&'static str>,
+    allowed_query_params: HashSet<&'static str>,
+}
+
+impl LoggingPolicy {
+    /// Create a new `LoggingPolicy`.
+    pub fn new(options: LoggingOptions) -> Self {
+        // Create owned HashSet from the defaults and extend with any additional entries
+        let mut allowed_headers: HashSet<&'static str> =
+            DEFAULT_ALLOWED_HEADER_NAMES.iter().copied().collect();
+        allowed_headers.extend(options.additional_allowed_header_names.iter().copied());
+
+        let mut allowed_query_params: HashSet<&'static str> =
+            DEFAULT_ALLOWED_QUERY_PARAMETERS.iter().copied().collect();
+        allowed_query_params.extend(options.additional_allowed_query_params.iter().copied());
+
+        Self {
+            allowed_headers,
+            allowed_query_params,
+        }
+    }
+}
+
+#[cfg_attr(target_arch = "wasm32", async_trait::async_trait(?Send))]
+#[cfg_attr(not(target_arch = "wasm32"), async_trait::async_trait)]
+impl Policy for LoggingPolicy {
+    async fn send(
+        &self,
+        ctx: &Context,
+        request: &mut Request,
+        next: &[Arc<dyn Policy>],
+    ) -> PolicyResult {
+        info!(
+            "==> OUTGOING REQUEST: url: {}, method: {}, headers: {{ {} }}",
```

Comment:
IMO, the format feels too foreign. We don't have anything like this now and they almost stick out too much. I actually don't mind the arrows, but maybe not all caps. Not a big issue, though. Just something to consider. They just otherwise look really different and almost scream at you when other traces might be more interesting.